scripts/o3de/tests/test_enable_gem.py

```diff
@@ -164,3 +190,152 @@ def add_gem_dependency(enable_gem_cmake_file: pathlib.Path, gem_name: str):
                 assert gem_json.get('gem_name', '') in project_json.get('gem_names', [])
             else:
                 assert gem_json.get('gem_name', '') not in project_json.get('gem_names', [])
+
+    @pytest.mark.parametrize("gem_version, gem_dependencies, gem_names_and_versions, check, force, "
+                             "compatible_engines, engine_api_dependencies, test_engine_name, test_engine_version, "
+                             "test_engine_api_versions, expected_result", [
+        # passes when no version information is provided
+        pytest.param(None, [], {}, False, False, [], None, 'o3de-test', '', None, 0),
+        pytest.param('', [], {}, False, False, [], None, 'o3de-test', '', None, 0),
+        # passes when a compatible engine version is provided
+        pytest.param('1.2.3', [], {}, False, False, ['o3de-test>=1.0.0'], [], 'o3de-test', '1.0.0', {}, 0),
+        pytest.param('1.2.3', [], {}, False, False, ['o3de-test>=1.0.0, <2.3.4'], [], 'o3de-test', '2.0.1', {}, 0),
+        pytest.param('1.2.3', [], {}, False, False, ['o3de-test'], [], 'o3de-test', '', {}, 0),
+        pytest.param('1.2.3', [], {}, False, False, ['o3de-test'], [], 'o3de-test', '1.2.3', {}, 0),
+        pytest.param('1.2.3', [], {}, False, False, ['o3de-test==1.2.3'], [], 'o3de-test', '', {}, 1),
+        pytest.param('1.2.3', [], {}, False, False, ['o3de-test==1.2.3'], [], 'o3de-test', None, {}, 1),
+        # fails when no compatible engine version is found
+        pytest.param('1.2.3', [], {}, False, False, ['o3de-test<1.0.0'], [], 'o3de-test', '1.0.0', {}, 1),
```

Comment:
The comment should be moved above these parameters since these are expected fail due to not finding a compatible engine version.